### PR TITLE
feat(connect): retry FW hash check

### PR DESCRIPTION
**Pull request by Lemonexe**
## Description

- retry FW hash check if it ended up in `other-error`
  - max 3 attempts
  - if it succeeds within 3 attempts, suite sends warning to Sentry with `lastErrorPayload` and `attempts` count (no. of successful attempt)
- cleanup the connect getFeatures code a little bit (consistent boilerplate with revision check)

[See testing branch](https://github.com/trezor/trezor-suite/tree/feat/fw-hash-check-retry-TESTING) to play around with simulated retried check :wink: 

### Notes :thought_balloon: 

This implementation is kinda hacky - it relies on the fact that Suite does call `Device.run()` multiple times for whatever reasons. I put no mechanism in place to ensure that FW hash check is repeated after an `other-error`; repeating it is only a side effect _when_ Suite calls `Device.run()`. If Suite wouldn't call again, hash check wouldn't be retried.

I think that's ok at the moment - we're just collecting diagnostics for `other-error`, and no UI is bound to it. We might find out that the retries are no longer necessary and abandon them. But if we do need them, we'd need to either:
1) ensure Suite calls `Device.run()` until (success || max retries), or
2) recursively call `_runInner` in from `_runInner` until (success || max retries)
_(but then we'd need some timeout maybe? To ensure the calls are not spammed right after each other - that'd be really futile if device is busy)._

## Related Issue

Resolve https://github.com/trezor/trezor-suite-private/issues/120

## Screenshots:

suite log from testing branch
[debug branch suite.txt](https://github.com/user-attachments/files/17979107/debug.branch.suite.txt)

connect log from testing branch
[debug branch connect.txt](https://github.com/user-attachments/files/17979109/debug.branch.connect.txt)

### CODE REVIEW

**Comment by mroz22** [comment]
related #15686

**Comment by mroz22** [comment]
@marekrjpolak, please, lets survive one more review day 🙏

File: packages/connect/src/device/Device.ts [hunk 1]
```diff
@@ -583,22 +583,8 @@ export class Device extends TypedEmitter<DeviceEvents> {
...
-
-        if (
-            // The check was not yet performed
-            this.authenticityChecks.firmwareRevision === null ||
-            // The check was performed, but outcome cannot be surely determined (Suite is offline or there was an unexpected error)
-            // -> recheck on every getFeatures() until the result is known
-            (this.authenticityChecks.firmwareRevision.success === false &&
-                ['cannot-perform-check-offline', 'other-error'].includes(
-                    this.authenticityChecks.firmwareRevision.error,
-                ))
```
Comment (Lemonexe) [inline packages/connect/src/device/Device.ts:602]: This condition is abstracted to `checkFirmwareRevisionWithRetries`. I think with named variables it doesn't need comments :slightly_smiling_face:

**Review by marekrjpolak — APPROVED** [review]
The Connect part looks reasonable.

File: packages/connect/src/device/Device.ts [hunk 2]
```diff
@@ -750,6 +736,31 @@ export class Device extends TypedEmitter<DeviceEvents> {
         this._updateFeatures(message);
     }
 
+    async checkFirmwareHashWithRetries() {
+        const lastResult = this.authenticityChecks.firmwareHash;
+        const notDoneYet = lastResult === null;
+        const attemptsDone = lastResult?.attempt ?? 0;
+        if (attemptsDone >= FIRMWARE.FW_HASH_CHECK_MAX_ATTEMPTS) return;
+
+        const retriableErrors = ['other-error'];
```
Comment (peter-sanderson) [inline packages/connect/src/device/Device.ts]: nit: maybe move to CASE_CONSTANT out of function block?

File: packages/connect/src/device/Device.ts [hunk 3]
```diff
@@ -818,6 +829,19 @@ export class Device extends TypedEmitter<DeviceEvents> {
         }
     }
 
+    async checkFirmwareRevisionWithRetries() {
+        const lastResult = this.authenticityChecks.firmwareRevision;
+        const notDoneYet = lastResult === null;
+
+        const retriableErrors = ['cannot-perform-check-offline', 'other-error'];
```
Comment (peter-sanderson) [inline packages/connect/src/device/Device.ts]: nit: extract to const ouside of function block

File: packages/connect/src/device/Device.ts (hunk 2 above)
Comment (Lemonexe) [reply packages/connect/src/device/Device.ts]: Good point. Now looking at it again, it's also untyped, I accidentally did `Array<string>.includes()`

File: packages/connect/src/types/device.ts [hunk 4]
```diff
@@ -65,8 +65,13 @@ export type FirmwareHashCheckError =
     | 'unknown-release'
     | 'other-error';
 export type FirmwareHashCheckResult =
-    | { success: true }
-    | { success: false; error: FirmwareHashCheckError; errorPayload?: unknown };
+    | { success: true; attempt?: number; warningPayload?: unknown }
```
Comment (peter-sanderson) [inline packages/connect/src/types/device.ts]: thinking laud: `attempt` -> `attemptCount`?

**Review by peter-sanderson — APPROVED** [review]

File: packages/connect/src/device/Device.ts (hunk 3 above)
Comment (Lemonexe) [reply packages/connect/src/device/Device.ts]: 23429449

File: packages/connect/src/device/Device.ts (hunk 2 above)
Comment (Lemonexe) [reply packages/connect/src/device/Device.ts]: 23429449

File: packages/connect/src/types/device.ts (hunk 4 above)
Comment (Lemonexe) [reply packages/connect/src/types/device.ts]: 23429449

**Pull request merged**